### DISCUSSION TITLE
Using phase4 in SBT (scala-based) environment

### DISCUSSION THREAD

**Discussion by chamikaCN** [phax/phase4#89] [Q&A] [answered]
Hi Philip,

I intend to use the phase 4 library to deliver messages via AS4 and I have already used the repository for feasibility testing successfully. My requirement is to use phase 4 to build the system in `scala/sbt based` environment. I tried to find out whether there are `sbt` versions of the phase4 maven repos available without success. 

Can you please let me know if such libraries are available? If not any plans to extend phase4 to `sbt` as well?

**Comment by phax** [comment]
Hi Chamika,
Unfortunately there is no Scala build of phase4 available yet - sorry.
I also don't have any particular knowledge that could bring your forward.

The phase4 builds currently require Java 1.8 or later to work. If you can create a PR with Scala support - happy to discuss this in further detail. All publications are currently only in Maven Central.

**Reply by chamikaCN** [reply]
What I actually meant was to ask whether `sbt` support available. Sorry that, I might have confused you with `scala` stuff. Actually, I can use this same codebase and libraries set in Scala projects as well since they support that. If the libraries were available in `sbt` the dependency management would be easier.

As a suggestion, phase4 might expand well among the scala community with `sbt` integration. As per my knowledge, it is a similar procedure to maven. No change in the codebase is required. BTW I'm not an expert in this area. Just a suggestion. :slightly_smiling_face: 

Thanks again for the quick response.

**Comment by chamikaCN** [comment, marked as answer]
For anyone who does not have much experience with Scala and SBT, I later found out that all Maven repositories can be imported to SBT projects without any difficulties. Just have to change the syntax to match SBT's `build.sbt` Further details can be found here. 

1. [how-use-maven-repository-library-with-scala-sbt](https://alvinalexander.com/scala/how-use-maven-repository-library-with-scala-sbt/)
2. [Official Documentation](https://www.scala-sbt.org/1.x/docs/Library-Dependencies.html)

HTH someone, especially new to the domain, as I  was. :smiley:

**Reply by phax** [reply]
Cool thanks @chamikaCN  :)
